src/pages/library/dataBrowser-utils.ts

```diff
@@ -76,98 +96,110 @@ export const getDataModalityListFromDataset = dataset => _.flow(
 )(dataset['prov:wasGeneratedBy'])
 
 
-export const getAssayCategoryListFromDataset = dataset => _.flow(
+export const getAssayCategoryListFromDataset = (dataset: DatasetResponse) => _.flow(
   _.flatMap('TerraCore:hasAssayCategory'),
   _.sortBy(_.toLower),
   _.compact,
   _.uniqBy(_.toLower)
-)(dataset['prov:wasGeneratedBy'])
-
-export const formatDatasetTime = time => !!time ? Utils.makeStandardDate(new Date(time)) : null
+)(dataset['prov:wasGeneratedBy']) as string[]
 
+export const formatDatasetTime = (time: string | null) => !!time ? Utils.makeStandardDate(new Date(time)) : null
 
-export const getDatasetAccessType = dataset => Utils.cond(
-  [isExternal(dataset), () => datasetAccessTypes.EXTERNAL],
-  [dataset.accessLevel === 'reader' || dataset.accessLevel === 'owner', () => datasetAccessTypes.GRANTED],
-  () => datasetAccessTypes.CONTROLLED)
+// Return type should be decided by above
+export const getDatasetAccessType = (dataset: DatasetResponse) => Utils.cond(
+  [isExternal(dataset), () => datasetAccessTypes.External],
+  [dataset.accessLevel === 'reader' || dataset.accessLevel === 'owner', () => datasetAccessTypes.Granted],
+  () => datasetAccessTypes.Controlled)
 
-export const DatasetAccess = ({ dataset }) => {
-  const { requestingAccess, setRequestingAccess } = useState()
+interface DatasetAccessProps {
+  dataset: Dataset
+}
+export const DatasetAccess = ({ dataset }: DatasetAccessProps) => {
+  const [requestingAccess, setRequestingAccess] = useState(false)
   const access = getDatasetAccessType(dataset)
   const { requestAccessURL } = dataset
   const buttonStyle = { height: 34, textTransform: 'none', padding: '.5rem' }
   const textStyle = { paddingLeft: 10, paddingTop: 4, fontSize: 12 }
 
   return h(Fragment, [
     Utils.cond(
-      [!!requestAccessURL && access === datasetAccessTypes.CONTROLLED, () => {
+      [!!requestAccessURL && access === datasetAccessTypes.Controlled, () => {
         return h(ButtonOutline, {
           style: buttonStyle,
           href: requestAccessURL, target: '_blank'
         }, [icon('lock'), div({ style: { paddingLeft: 10, fontSize: 12 } }, ['Request Access'])])
       }],
-      [access === datasetAccessTypes.CONTROLLED, () => h(ButtonOutline, {
+      [access === datasetAccessTypes.Controlled, () => h(ButtonOutline, {
         style: buttonStyle,
         onClick: () => {
-          setRequestingAccess()
+          setRequestingAccess(true)
+          // @ts-expect-error
```

Comment:
Should we have a convention to add an explaining comment whenever we use `ts-expect-error`? That's what I'd expect if this was Java and the code was using `SuppressWarnings`.

Looking at the typescript docs it sounds like the common use of this form is for test code. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments